source/app-development/tutorial-ps-to-quota.rst

```diff
@@ -0,0 +1,451 @@
+   * - test/test_command.rb
+     - a unit test of the parsing code
+   * - views/index.html
+     - the main section of the html page template using an implementation of `ERB <https://ruby-doc.org/stdlib-2.2.0/libdoc/erb/rdoc/ERB.htm://ruby-doc.org/stdlib-2.2.0/libdoc/erb/rdoc/ERB.html>`__
+       called `erubi <https://github.com/jeremyevans/erub://github.com/jeremyevans/erubi>`__
+       which auto-escapes output of ERB tags by default (for security)
+   * - views/layout.html
+     - the rendered HTML from views/index.html is insterted into this layout,
+       where css and javascript files are included
+
+.. list-table:: Other files
+   :header-rows: 1
+
+   * - File
+     - Description
+   * - Gemfile, Gemfile.lock
+     - defines gem dependencies for the app (see `Bundler's Rationale <http://bundler.io/rationale.html>`__)
+   * - vendor/bundle
+     - installed and versioned gem dependencies
+   * - tmp/
+     - tmp directory is kept so its easier to ``touch tmp/restart.txt`` when you
+       want to force Passenger to restart an app
+   * - public/
+     - serve up static assets like Bootstrap css; in OnDemand, NGINX auto-serves
+       all files under public/ directly, without going through the Passenger
+       process, which makes this much faster; as a result, each static file is
+       in a directory with an explicit version number, so if these files ever
+       change we change the version, which is one cache busting strategy
+   * - Rakefile
+     - this provides a default Rake task for running the automated tests under
+       test/, so you can run the tests by running the command ``rake``
+   * - test/minitest_helper.rb
+     - contains setup code common between all tests
+
+Clone and setup
+---------------
+
+#. Login to Open OnDemand, click "Develop" dropdown menu and click the "My Sandbox Apps (Development)" option.
+#. Click "New Product" and "Clone Existing App".
+#. Fill out the form:
+
+   #. Directory name: quota
+   #. Git remote: https://github.com/OSC/ood-example-ps
+   #. Check "Create new Git Project from this?"
+   #. Click Submit
+
+#. Launch the app by clicking the large blue Launch button. In a new browser
+   window/tab you will see the output of a ps command filtered using grep.
+
+#. Switch browser tab/windows back to the dashboard Details view of the app and
+   click "Files" button on the right to open the app's directory in the File
+   Explorer.
+
+
+Edit to run and parse quota
+---------------------------
+
+The app runs and parses this command:
+
+.. code:: sh
+
+   ps aux | grep '[A]pp'
+
+We will change it to run and parse this command:
+
+.. code:: sh
+
+   quota -spw
+
+Update test/test_command.rb
+...........................
+
+Run the command to get example data. Copy and paste the output into the test, and
+update the assertions to expect an array of "quotas" instead of "processes"
+with appropriate attributes.
+
+Diff:
+
+.. code:: diff
```

Comment:
Sphinx doesn't seem to handle this very well. I wonder if there's a different code for github markdown style diffs?

![diff](https://user-images.githubusercontent.com/2374718/30763327-ae40cd20-9fb3-11e7-8046-8f890187b3d2.png)

Reply:
Try using...

```rst
.. code-block:: diff

   stuff...
```

instead of `code`.